### ISSUE TITLE
Store PHP exec path in environment var for script access

### ISSUE THREAD

**Issue by dccampbell** [composer/composer#7850]
**Summary**: Make the current PHP executable accessible to composer scripts via an environment variable.

**Details**: One of the biggest limitations of composer scripts is the inability to use references except at the very beginning, which makes it impossible to do chaining (`@build && @deploy || @error`) or pass paths as arguments (`build.sh --PHP_EXEC=@php`). There are good technical reasons for this.

However, these limitations can be worked around as long as a script has the ability to call/reference the php and composer that were used to invoke it. Currently, the `COMPOSER_BINARY` environment variable is already set when scripts are running. Unfortunately, PHP itself is not so accessible, which forces scripts to rely on `PATH` or add their own complex exec finding logic. In my experience, any composer script relying on `PATH` this way is fragile and will inevitably fail in someone's environment.

**Suggestion**: Simply set an environment variable (`PHP_BINARY`?) to the same path that `@php` uses, such that scripts can utlize it.

**Comment by Seldaek** [comment]
PHP_BINARY is set by PHP itself since PHP 5.4, I suggest you use that :)

**Comment by dccampbell** [comment]
> PHP_BINARY is set by PHP itself since PHP 5.4, I suggest you use that :)

@Seldaek Right, but I'm talking about the _shell_ environment. So the issue is specifically that I _can't_ use that. Sorry if that wasn't clear. 

Currently, you can only get `PHP_BINARY` in composer scripts that start with `@php`. For example, calling `sh build.sh` leaves you with no reliable way to actually use PHP. I can't even safely call composer itself, because even though `COMPOSER_BINARY` is provided, if I call it w/o specifying the PHP exec then I could end up with a completely different PHP version than was used to originally invoke the composer script.

So basically PHP's `PHP_BINARY` just needs to get passed into the environment for composer scripts.

**Comment by Seldaek** [comment]
Ok I see, you have shell scripts being called, I indeed didn't get that. I guess we could export the PHP_BINARY const as an env var before we execute shell scripts.

**Comment by Seldaek** [comment]
Fixed by https://github.com/composer/composer/pull/7857
